client/rest-high-level/src/test/java/org/elasticsearch/client/CrudIT.java

```diff
@@ -141,4 +144,40 @@ public void testGet() throws IOException {
             assertEquals("value1", sourceAsMap.get("field1"));
         }
     }
+
+    public void testDelete() throws IOException {
+        {
+            DeleteRequest deleteRequest = new DeleteRequest("index", "type", "does_not_exist");
+            DeleteResponse deleteResponse = execute(deleteRequest, highLevelClient()::delete, highLevelClient()::deleteAsync);
+            assertEquals("index", deleteResponse.getIndex());
+            assertEquals("type", deleteResponse.getType());
+            assertEquals("does_not_exist", deleteResponse.getId());
+            assertEquals(DocWriteResponse.Result.NOT_FOUND, deleteResponse.getResult());
+            assertEquals(1, deleteResponse.getVersion());
```

Comment:
I think Get request returns an error with NOT FOUND response code in this case? Which make more sense to me. 

Anyway a delete increments the version even if the document is not found, and is kept in the versions map as a tombstone.